src/Component/InlineEdit.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * A Simple inline editable text Vue component.
+ */
+
+namespace atk4\ui\Component;
+
+use atk4\ui\Exception;
+use atk4\ui\jsVueService;
+use atk4\ui\View;
+
+class InlineEdit extends View
+{
+    public $defaultTemplate = 'inline-edit.html';
+
+    /**
+     * jsCallback for saving data.
+     *
+     * @var null
+     */
+    public $cb = null;
+
+    /**
+     * Whether callback should save value to db automatically or not.
+     * Default to using onChange handler.
+     * If set to true, then saving to db will be done when model get set
+     * and if model is loaded already.
+     *
+     * @var bool
+     */
+    public $autoSave = false;
+
+    /**
+     * The actual db field name that need to be saved.
+     * Default to title field when model is set.
+     *
+     * @var null|string The name of the field.
+     */
+    public $modelField = null;
+
+    /**
+     * Whether component should save it's value when input get blur.
+     *
+     * @var bool
+     */
+    public $saveOnBlur = true;
+
+    /**
+     * Default css for the input div.
+     *
+     * @var string
+     */
+    public $inputCss = 'ui right icon input';
```

Comment:
normally we use public $ui = 'right icon input' 

this is inconsistent with other views.

Reply:
This is not the regular css template field {$_class} or {$class} which define the view that create the Vue instance. The inputCss value is for defining the inline edit input inside the Vue template.
I have set it up as a user defined value. This is how you would change the input look of the inline edit field.